activation/post.go

```diff
@@ -169,23 +170,32 @@ func DefaultPostSetupOpts() PostSetupOpts {
 		DataDir:          opts.DataDir,
 		NumUnits:         opts.NumUnits,
 		MaxFileSize:      opts.MaxFileSize,
-		ProviderID:       opts.ProviderID,
+		ProviderID:       math.MaxUint64,
 		Throttle:         opts.Throttle,
 		Scrypt:           opts.Scrypt,
 		ComputeBatchSize: opts.ComputeBatchSize,
 	}
 }
 
-func (o PostSetupOpts) ToInitOpts() config.InitOpts {
-	return config.InitOpts{
+func (o PostSetupOpts) ToInitOpts() (config.InitOpts, error) {
+	if o.ProviderID == math.MaxUint64 {
+		return config.InitOpts{}, fmt.Errorf("no provider ID specified: %d", o.ProviderID)
+	}
```

Comment:
Why use an uin64 and a magic number `math.MaxUint64` to express optionality instead of `*uint32` as in the post repo?